### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by Melisa-Zorraindo**
I'm finally done with this. I had to spend a few hours with the `setFilteredVenues` in the `VenueContext` because it would end up in an infinite loop no matter what. 

Everything you see inside the `useEffect()` in the Sidebar component was inside its own function `filterProperties()`. I would call this function from the `useEffect()`, but I got a `Maximum update depth exceeded warning` every time. I tried wrapping the function in a `useCallback()` (which was the VSCode recommended method), but still had the same issue. I could finally solve it by moving the code inside the `useEffect()`.

This is what the filters look like:
image
The cards are still the old ones, don't pay attention to them. They're in the picture to provide context.

### CODE REVIEW

File: src/components/Sidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    3: false,
+    2: false,
+    1: false,
+    0: false,
+  });
+  const [maxGuests, setMaxGuests] = useState<number>(0);
+
+  //initial price values
+  const sortedVenues = allvenues.sort((a, b) => b.price - a.price);
+  const mostExpensive = sortedVenues[0]?.price;
```
Comment (chrislewispac) [inline src/components/Sidebar/index.tsx:42]: is having no venues a valid state for your application? if so, how are you handling it?

File: src/components/Sidebar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  //filter functionality
+  useEffect(() => {
+    const filteredProperties = allvenues.filter((venue) => {
+      let featureMatches;
+      let ratingMatches;
+      let guestsMatches;
+      let minPriceMatches;
+      let maxPriceMatches;
+
+      if (filters.features.length !== 0) {
```
Comment (chrislewispac) [inline src/components/Sidebar/index.tsx:120]: let's set up a call to do some practice on how to make this more abstract/extensible!

**Review by chrislewispac — APPROVED** [review]
Im glad you got this working but let's set up a call to discuss a more flexible way to handle this!

File: src/components/Sidebar/index.tsx (hunk 1 above)
Comment (Melisa-Zorraindo) [reply src/components/Sidebar/index.tsx:42]: If the API call fails the user will receive feedback

File: src/components/Sidebar/index.tsx (hunk 2 above)
Comment (Melisa-Zorraindo) [reply src/components/Sidebar/index.tsx:120]: Done

**Pull request merged**